.evergreen/config.yml

```diff
@@ -827,20 +827,18 @@ functions:
 
   start-kms-mock-server:
     - command: shell.exec
-      type: test
       params:
-        working_dir: src
-        background: true
         script: |
           ${PREPARE_SHELL}
+
           cd ${DRIVERS_TOOLS}/.evergreen/csfle
-          cat <<EOF > kms_setup.json
-          {
-              "kms_ca_file": "${KMS_CA_FILE}",
-              "kms_cert_file": "${KMS_CERT_FILE}"
-          }
-          EOF
-          mongo --nodb mock_kms.js
+          . ./activate_venv.sh
+    - command: shell.exec
```

Comment:
Much like the mock OCSP functions, the first command sets up the local environment in the foreground, and the second command starts the Python mock server in the background. These need to be separated for the tests to consistently find the mock KMS server.

Reply:
Interesting. https://github.com/evergreen-ci/evergreen/wiki/Project-Commands#shellexec notes:

> background: if set to true, does not wait for the script to exit before running the next commands

My new hypothesis for the cause of the connection refused errors:
- Evergreen would run `start-kms-mock-server` and proceed before the script completed.
- The Go driver tests started before the mock KMS server started.

Starting the virtual environment in a non-background command before helps. But I think this is still hiding a race. 

If the mock KMS server does not establish listening sockets before the Go driver tests run, I suspect the same issue will occur. But, given that the OCSP tasks have a similar setup, I bet the likelihood of the KMS server not starting before the Go tests run is slim to none. If we see it failing in the future, we could consider appending a foreground command to loop until it can establish a connection on port 8000. That seems unnecessary for now.

Reply:
That sounds exactly right. I think all the current mock servers in testing (KMS, OCSP and maybe load balancer?) have this racey behavior. It seems that if you only have the server-starting call in the `background` function, the tests pretty much never start before the server. So, if we start to see failures we can consider something like a foreground loop.